plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/TransformPom.java

```diff
@@ -0,0 +1,58 @@
+package org.jenkins.tools.test.hook;
+
+import org.jenkins.tools.test.model.MavenCoordinates;
+import org.jenkins.tools.test.model.MavenPom;
+import org.jenkins.tools.test.model.PomData;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class TransformPom extends PluginCompatTesterHookBeforeExecution {
```

Comment:
This is the kind of transformation I think we should pursue. Have the different possible operations encapsulated as strategies and get the different plugins customize the flow (providing extra steps if needed). A default flow (equivalent to today's), would be use for every plugin not providing customization.